herbert_core/utilities/classes/@data_op_interface/binary_op_manager.m

```diff
@@ -111,12 +111,17 @@
     % w1 is a double array; w2 must have class 'classname'
     if ~isscalar(w1)
         size_stack1 = size(w2);
-        [size_root1, ok] = size_array_split (size(w1), size(w2));
-        if ~ok
-            error('HERBERT:data_op_interface:invalid_argument', ...
-                ['Unable to resolve the numeric array into a stack of arrays, ',...
-                'with stack size matching the object array size.']);
+
+        try
+            size_root1 = size_array_split (size(w1), size(w2));
+        catch ME
+            err = MException('HERBERT:data_op_interface:invalid_argument', ...
+                             ['Unable to resolve the numeric array into a stack of arrays, ',...
+                              'with stack size matching the object array size.']);
```

Comment:
I do not think you need to catch/rethrow the error here as `size_array_split` informs about the issue well enough. Checks and exception here only hide the reason. Better then before but just throwing would be clearler (and shorter)

The same opinion about couple of places below